bn-apps/ledger-sync/ledger-sync-service/src/main/kotlin/net/corda/businessnetworks/ledgersync/RequestLedgersSyncFlow.kt

```diff
@@ -0,0 +1,63 @@
+package net.corda.businessnetworks.ledgersync
+
+import co.paralleluniverse.fibers.Suspendable
+import net.corda.core.crypto.SecureHash
+import net.corda.core.flows.FlowLogic
+import net.corda.core.flows.FlowSession
+import net.corda.core.flows.InitiatedBy
+import net.corda.core.flows.InitiatingFlow
+import net.corda.core.identity.Party
+import net.corda.core.serialization.CordaSerializable
+import net.corda.core.utilities.unwrap
+
+/**
+ * If inconsistencies are flagged, the counter party can be queried for a more detailed report using this flow. Based
+ * the result, both the parties can take further action (such as notifying the BNO) or recover the transactions found.
+ */
+@InitiatingFlow
+class RequestLedgersSyncFlow(
+        private val members: List<Party>
+) : FlowLogic<Map<Party, LedgerSyncFindings>>() {
+
+    @Suspendable
+    override fun call(): Map<Party, LedgerSyncFindings> = (members - ourIdentity)
+            .map { they ->
+                val knownTransactionsIds = serviceHub.vaultService.withParticipants(ourIdentity, they)
+                val findings = initiateFlow(they).sendAndReceive<LedgerSyncFindings>(knownTransactionsIds).unwrap { it }
+                // an individual implementation might treat findings individually
+                they to findings
+            }.toMap()
+}
+
+@Suppress("unused")
+@InitiatedBy(RequestLedgersSyncFlow::class)
+class RespondLedgerSyncFlow(
+        private val otherSideSession: FlowSession
+) : FlowLogic<Unit>() {
+
+    /**
+     * Depending on the use case, users of this flow might introduce additional validation logic. This could mean
+     * ensuring the requester is still a valid member of a network at the time of request or similar checks.
+     */
+    @Suspendable
+    override fun call() {
+        val theirs = otherSideSession.receive<List<SecureHash>>().unwrap { it }
```

Comment:
Might make sense to add a commented block of code, showing how this verification could be done using the business network membership service to make it more xplicit